src/encoded/audit/experiment.py

```diff
@@ -2436,16 +2436,28 @@ def audit_experiment_control(value, system, excluded_types):
         return
 
     for control in value['possible_controls']:
-        if control.get('biosample_term_id') != value.get('biosample_term_id'):
-            detail = 'The specified control {} for this experiment is on {}, '.format(
-                control['@id'],
-                control.get('biosample_term_name')) + \
-                'but this experiment is done on {}.'.format(
-                    value['biosample_term_name'])
+        if not is_matching_biosample_control(control, value.get('biosample_term_id')):
+            detail = ('The specified control {} '
+                      'for this experiment is on {}, '
+                      'but this experiment is done on {}.').format(
+                        control['@id'],
+                        control.get('biosample_term_name'),
+                        value['biosample_term_name']
+                      )
             yield AuditFailure('inconsistent control', detail, level='ERROR')
     return
 
 
+def is_matching_biosample_control(dataset, biosample_term_id):
+    if (dataset['@type'][0] == 'Experiment'):
+        return dataset.get('biosample_term_id') == value.get('biosample_term_id')
```

Comment:
You are calling this function on line 2439: `is_matching_biosample_control(control, value.get('biosample_term_id'))` and inside the function returning `dataset.get('biosample_term_id') == value.get('biosample_term_id')` Am I missing something? Where is the value variable coming from in the function?